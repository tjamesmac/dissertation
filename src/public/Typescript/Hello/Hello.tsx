import * as React from 'react';
const style = require('../../Scss/home.scss');
export interface IHelloProps { name: string; }

const fetchGraphQL = async () => {
  try {
    const query = `
    query {
      hello
    }
    `;

    const url = 'http://localhost:3000/graphql';
    const options = {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ query }),
    };

    const fetchData = await fetch(url, options);
    const response = await fetchData;
    if (response.status === 200) {
      const responseJson = await response.json();
      const responseData = responseJson.data.hello;
      console.log(responseData);
      const target = document.getElementById('response');
      const p = document.createElement('p');
      const text = document.createTextNode(responseData);
      p.appendChild(text);
      if (target !== null) {
        target.appendChild(p);
      }
      console.log(responseJson);
    }

  } catch (error) {
    console.error('this is my error', error);
  }
};

export const Hello: React.FunctionComponent<{name: string}> = (props: IHelloProps) => {
  fetchGraphQL();
  return (
    <div>
      <h1 className='header'>Hello, {props.name}. Welcome back!</h1>
      <div id='response' className='response'>

      </div>
    </div>
  );
};
