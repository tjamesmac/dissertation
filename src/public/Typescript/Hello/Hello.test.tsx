import { shallow } from 'enzyme';
import * as React from 'react';
import renderer from 'react-test-renderer';
import { Hello } from './Hello';

describe('Test Hello component', () => {
  it('renders the heading with a name', () => {
    const result = shallow(<Hello name='Test'/>).contains(
    <div>
      <h1 className='header'>Hello, Test. Welcome back!</h1>
    </div>);
    expect(result).toBeTruthy();
  });
});
