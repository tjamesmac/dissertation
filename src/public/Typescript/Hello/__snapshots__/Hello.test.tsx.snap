// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component it renders correctly 1`] = `
<div>
  <h1
    className="header"
  >
    Hello, 
    Test
    . Welcome back!
  </h1>
</div>
`;
